File: server/src/knowledge/ingestors/pptxIngestor.js
import { BaseIngestor } from '../baseIngestor.js';

// Needs a pptx text-extraction lib (e.g. `pptx-parser` or manual zip/XML walk).
export class PptxIngestor extends BaseIngestor {
  constructor() {
    super('document');
  }

  async ingest(_source) {
    throw new Error('PptxIngestor not implemented — slide-by-slide text + speaker notes extraction needed');
  }
}

File: server/app/knowledge/ingestors/pptx_ingestor.py
from ..base_ingestor import BaseIngestor


class PptxIngestor(BaseIngestor):
    def __init__(self):
        super().__init__("document")

    async def ingest(self, source: dict) -> list[dict]:
        raise NotImplementedError("PptxIngestor not implemented — slide-by-slide text + speaker notes extraction needed")
